src/charm.py

```diff
@@ -66,439 +53,29 @@ def __init__(self, *args):
-                    f"Configuration options missing: {missing_config_options}"
-                )
-                return
-            if not self._config_certificates_are_valid:
-                self.unit.status = BlockedStatus("Certificates are not valid")
-                return
-            if self.unit.is_leader():
-                self._store_certificates_from_config()
-                self.tls_certificates.revoke_all_certificates()
-                logger.info("Revoked all previously issued certificates.")
-        self.unit.status = ActiveStatus()
-
-    def _store_certificates_from_config(self) -> None:
-        """Stores user provided certificates based on config values."""
-        ca_chain_config = self.model.config.get("ca-chain", None)
-        certificate_config = self.model.config.get("certificate", None)
-        ca_certificate_config = self.model.config.get("ca-certificate", None)
-        if not certificate_config:
-            raise ValueError("Config `certificate` not set")
-        if not ca_certificate_config:
-            raise ValueError("Config `ca-certificate` not set")
-        self._store_config_ca_certificate(
-            base64.b64decode(ca_certificate_config).decode("utf-8").strip()
-        )
-        self._store_config_certificate(
-            base64.b64decode(certificate_config).decode("utf-8").strip()
-        )
-        if ca_chain_config:
-            ca_chain_list = parse_ca_chain(base64.b64decode(ca_chain_config).decode("utf-8"))
-        else:
-            ca_chain_list = [
-                base64.b64decode(ca_certificate_config).decode("utf-8").strip(),
-                base64.b64decode(certificate_config).decode("utf-8").strip(),
-            ]
-        self._store_config_ca_chain(ca_chain_list)
-
-    def _generate_self_signed_certificates(self, certificate_signing_request: str) -> str:
-        """Generates self-signed certificates.
-
-        Args:
-            certificate_signing_request (str): Certificate signing request
-
-        Returns:
-            str: Certificate
-        """
-        if not self._self_signed_ca_private_key:
-            raise ValueError("CA Private key not stored")
-        if not self._self_signed_ca_private_key_password:
-            raise ValueError("CA private key password not stored")
-        if not self._self_signed_ca_certificate:
-            raise ValueError("CA Certificate not stored")
-        certificate = generate_certificate(
-            ca=self._self_signed_ca_certificate.encode(),
-            ca_key=self._self_signed_ca_private_key.encode(),
-            ca_key_password=self._self_signed_ca_private_key_password.encode(),
-            csr=certificate_signing_request.encode(),
-            validity=self._certificate_validity,
-        )
-        logger.info("Generated self-signed certificates")
-        return certificate.decode()
+        self.unit.status = WaitingStatus("Waiting for certificate creation request.")
```

Comment:
Alright, I agree, the charm itself is ready, the idea of putting in waiting status was to avoid running actions before any requirer has joined or requested a certificate creation.
But at the end I agree that it doesn't reflect the actual status of this charm.